core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2131,15 +2143,21 @@ class Log(@volatile var dir: File,
    *
    * @throws IOException if the file can't be renamed and still exists
    */
-  private def asyncDeleteSegment(segment: LogSegment) {
+  private def deleteSegmentFiles(segment: LogSegment, asyncDelete: Boolean = true) {
     segment.changeFileSuffixes("", Log.DeletedFileSuffix)
     def deleteSeg() {
       info(s"Deleting segment ${segment.baseOffset}")
       maybeHandleIOException(s"Error while deleting segments for $topicPartition in dir ${dir.getParent}") {
         segment.deleteIfExists()
       }
     }
-    scheduler.schedule("delete-file", deleteSeg _, delay = config.fileDeleteDelayMs)
+
+    if (asyncDelete) {
+      info(s"Scheduling log segment [baseOffset ${segment.baseOffset}, size ${segment.size}] for deletion.")
```

Comment:
Do we need this info logging given the logging in deleteSeg()?

Reply:
I think this one says when it's been scheduled versus when deletion started. Is that not useful?

Reply:
Hmm, it carries the same info just a bit early. Would that be useful?

Reply:
It could be useful if the scheduler gets overwhelmed and there's a long delay between scheduling and actual deletion. I don't feel strongly though, so leave it to you and @dhruvilshah3 to decide.

Reply:
I don't feel strongly either. Most of the times, we care about when a segment was deleted. When the related files were deleted is perhaps not as interesting to know. In that case, perhaps we could log every time a segment is removed from the `segments` map?